x/ecocredit/server/core/mint_batch_credits_test.go

```diff
@@ -0,0 +1,179 @@
+package core
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/orm/types/ormerrors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"gotest.tools/v3/assert"
+
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
+	"github.com/regen-network/regen-ledger/types/math"
+	"github.com/regen-network/regen-ledger/x/ecocredit/core"
+	"github.com/regen-network/regen-ledger/x/ecocredit/server/utils"
+)
+
+func TestMintBatchCredits_Valid(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	ctx := s.ctx
+	batch := setupMintBatchTest(s, true)
+
+	balBefore, err := s.stateStore.BatchBalanceTable().Get(ctx, s.addr, batch.Key)
+	assert.NilError(t, err)
+	supplyBefore, err := s.stateStore.BatchSupplyTable().Get(ctx, batch.Key)
+	assert.NilError(t, err)
+
+	mintTradable, mintRetired := math.NewDecFromInt64(10), math.NewDecFromInt64(10)
+	msg := core.MsgMintBatchCredits{
+		Issuer:     s.addr.String(),
+		BatchDenom: batch.Denom,
+		Issuance: []*core.BatchIssuance{
+			{Recipient: s.addr.String(), TradableAmount: mintTradable.String(), RetiredAmount: mintRetired.String(), RetirementJurisdiction: "US-OR"},
+		},
+		OriginTx: &core.OriginTx{
+			Typ: "Ethereum",
+			Id:  "210985091248",
+		},
+		Note: "bridged credits",
+	}
+
+	_, err = s.k.MintBatchCredits(ctx, &msg)
+	assert.NilError(t, err)
+
+	balAfter, err := s.stateStore.BatchBalanceTable().Get(ctx, s.addr, batch.Key)
+	assert.NilError(t, err)
+	supplyAfter, err := s.stateStore.BatchSupplyTable().Get(ctx, batch.Key)
+	assert.NilError(t, err)
+
+	assertCreditsMinted(t, balBefore, balAfter, supplyBefore, supplyAfter, mintTradable, mintRetired, math.NewDecFromInt64(0), 6)
+}
+
+// tests that batch credits can be minted and sent to an account with no previous balance
+func TestMintBatchCredits_MintToNewAccount(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	ctx := s.ctx
+	batch := setupMintBatchTest(s, true)
+
+	newAcc := sdk.AccAddress("NewAccount")
+	balBefore, err := s.stateStore.BatchBalanceTable().Get(ctx, newAcc, batch.Key)
+	assert.ErrorContains(t, err, ormerrors.NotFound.Error())
+	balBefore = &api.BatchBalance{BatchKey: batch.Key, Address: newAcc}
```

Comment:
No sure if it's necessary to test balance not found given we create the account in the line above.

```suggestion
	newAcc := sdk.AccAddress("NewAccount")
	balBefore := &api.BatchBalance{BatchKey: batch.Key, Address: newAcc}
```